fsd_utils/services/aws_extended_client.py

```diff
@@ -0,0 +1,349 @@
+import json
+from datetime import datetime
+
+import boto3
+from fsd_utils.services.aws_sqs_extended_client_exception import ExceptionMessages
+from fsd_utils.services.aws_sqs_extended_client_exception import (
+    SQSExtendedClientException,
+)
+from fsd_utils.services.aws_sqs_extended_client_util import check_message_attributes
+from fsd_utils.services.aws_sqs_extended_client_util import get_s3_key
+from fsd_utils.services.aws_sqs_extended_client_util import MESSAGE_POINTER_CLASS
+from fsd_utils.services.aws_sqs_extended_client_util import RESERVED_ATTRIBUTE_NAME
+from fsd_utils.services.aws_sqs_extended_client_util import validate_messages
+
+
+class SQSExtendedClient:
+    def __init__(
+        self,
+        aws_access_key_id=None,
+        aws_secret_access_key=None,
+        region_name="eu-west-2",
+        endpoint_url=None,
+        large_payload_support=None,
+        always_through_s3=None,
+        delete_payload_from_s3=None,
+        logger=None,
+        **kwargs,
+    ):
+        self.large_payload_support = large_payload_support
+        self.always_through_s3 = always_through_s3
+        self.delete_payload_from_s3 = delete_payload_from_s3
+        self.logger = logger
+
+        if aws_access_key_id and aws_secret_access_key:
+            self.sqs_client = boto3.client(
+                "sqs",
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+                **kwargs,
+            )
+            self.s3_client = boto3.client(
+                "s3",
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+            )
+        else:
+            """
+            if 'aws_access_key_id' and 'aws_access_key_id' are not provided make sure to provide
+            'AWS_ACCESS_KEY_ID' and 'AWS_SECRET_ACCESS_KEY' with environment variables
+            """
+            self.sqs_client = boto3.client(
+                "sqs",
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+                **kwargs,
+            )
+            self.s3_client = boto3.client(
+                "s3",
+                region_name=region_name,
+                endpoint_url=endpoint_url,
+            )
+
+    def get_queues(self, prefix=None):
+        """
+        Gets a list of SQS queues. When a prefix is specified, only queues with names
+        that start with the prefix are returned.
+
+        :param prefix: The prefix used to restrict the list of returned queues.
+        :return: A list of Queue names.
+        """
+        if prefix:
```

Comment:
can we move these if else to a small functions. would be better if use just use "if" easy to read and good practice

def get_queue_urls(self, prefix=None):
    if prefix:
       return self.sqs_client.list_queues(QueueNamePrefix=prefix)["QueueUrls"]
   return self.sqs_client.list_queues()["QueueUrls"]

queues = get_queue_urls(self, prefix=None)

Reply:
Fixed the changes